nuxeo-distribution/nuxeo-server-hotreload-tests/src/test/java/org/nuxeo/ftest/server/hotreload/HotReloadTestRule.java

```diff
@@ -0,0 +1,125 @@
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Kevin Leturc <kleturc@nuxeo.com>
+ *
+ */
+package org.nuxeo.ftest.server.hotreload;
+
+import static org.junit.Assert.fail;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Optional;
+import java.util.function.Function;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.junit.rules.TestWatcher;
+import org.junit.runner.Description;
+import org.nuxeo.ecm.core.api.NuxeoException;
+import org.nuxeo.functionaltests.RestHelper;
+
+/**
+ * This Rule gives mechanism for hot reload tests.
+ * 
+ * @since 9.3
+ */
+public class HotReloadTestRule extends TestWatcher {
+
+    public static final String NUXEO_RELOAD_PATH = "/sdk/reload";
+
+    private static final Log log = LogFactory.getLog(HotReloadTestRule.class);
+
+    protected final static Function<URL, URI> URI_MAPPER = url -> {
+        try {
+            return url.toURI();
+        } catch (URISyntaxException e) {
+            throw new NuxeoException("Unable to map the url to uri", e);
+        }
+    };
+
+    @Override
+    protected void starting(Description description) {
+        RestHelper.logOnServer(String.format("Starting test '%s#%s'", description.getTestClass().getSimpleName(),
+                description.getMethodName()));
+        deployDevBundle(description);
+    }
+
+    @Override
+    protected void finished(Description description) {
+        String className = description.getTestClass().getSimpleName();
+        String methodName = description.getMethodName();
+        RestHelper.logOnServer(String.format("Ending test '%s#%s'", className, methodName));
+        RestHelper.cleanup();
+        // reset dev.bundles file
+        updateDevBundles("# AFTER TEST: " + methodName);
+        RestHelper.logOnServer(String.format("Test ended '%s#%s'", className, methodName));
+    }
+
+    /**
+     * Deploys the dev bundle located under src/test/resources/${YOUR_TEST_CLASS_NAME}/${YOUR_TEST_NAME} (if exists).
+     * <p />
+     * This method use {@link #updateDevBundles(String)}.
+     */
+    protected void deployDevBundle(Description description) {
+        Class<?> testClass = description.getTestClass();
+        // first lookup the absolute paths
+        String className = testClass.getSimpleName();
+        String methodName = description.getMethodName();
+        String relativeBundlePath = "/" + className + "/" + methodName;
+        // #getResource could return null if resource doesn't exist
+        Optional<String> bundlePath = Optional.ofNullable(testClass.getResource(relativeBundlePath))
+                                              .map(URI_MAPPER)
+                                              .map(Paths::get)
+                                              .map(Path::toAbsolutePath)
```

Comment:
Indeed, this could be interesting if we share these functions somewhere. Do we want that ? For sure it's an interesting topic to address if we introduce more and more functional code.